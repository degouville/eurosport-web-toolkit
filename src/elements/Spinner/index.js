import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'react-emotion';
import * as colors from '../../colors';

const rotate = keyframes`
to {
    transform: rotate(360deg);
  }
`;

const StyledWrapper = styled.div`
  margin: 10px;
  width: ${props => props.width};
  height: ${props => props.width};
`;

const Spinner = ({ color, width, className }) => (
  <StyledWrapper width={width} className={className}>
    <svg
      viewBox="0 0 66 66"
      xmlns="http://www.w3.org/2000/svg"
      css={css`
        animation: ${rotate} 2s linear infinite;
      `}
    >
      <circle
        style={{
          fill: 'transparent',
          strokeWidth: 2,
          width: '100%',
          height: '100%',
          strokeDasharray: 170,
          strokeDashoffset: 20,
        }}
        cx="33"
        cy="33"
        r="31"
        stroke="url(#gradient)"
      />
      <linearGradient id="gradient">
        <stop offset="50%" stopColor={color} stopOpacity="1" />
        <stop offset="65%" stopColor={color} stopOpacity="1" />
        <stop offset="100%" stopColor={color} stopOpacity="0" />
      </linearGradient>

      <circle
        style={{ strokeDasharray: 170, strokeDashoffset: 20 }}
        cx="36.8"
        cy="2.3"
        r="0.5"
        fill={color}
        stroke={color}
      />
    </svg>
  </StyledWrapper>
);

Spinner.defaultProps = {
  color: colors.coreLightMinus1,
  className: undefined,
};

Spinner.propTypes = {
  color: PropTypes.string,
  width: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Spinner;
