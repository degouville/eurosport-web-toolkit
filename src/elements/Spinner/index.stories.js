import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { Spinner } from '../..';

const spinnerStories = storiesOf('Elements|Spinner', module);

spinnerStories.add('Spinner', () => <Spinner color={text('color', '#fff')} width={text('width', '80px')} />);
