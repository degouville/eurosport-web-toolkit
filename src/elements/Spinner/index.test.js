import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '.';

describe('Spinner', () => {
  it('should render default spinner', () => {
    const wrapper = shallow(<Spinner className="test" width="50px" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with color passed as prop', () => {
    const wrapper = shallow(<Spinner className="test" color="red" width="50px" />);
    expect(
      wrapper
        .find('stop')
        .first()
        .prop('stopColor')
    ).toEqual('red');
  });

  it('should render with size passed as prop', () => {
    const wrapper = shallow(<Spinner className="test" color="red" width="100px" />);
    expect(wrapper.first()).toHaveStyleRule('height', '100px');
  });
});
