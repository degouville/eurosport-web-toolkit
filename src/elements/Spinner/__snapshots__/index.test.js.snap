// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render default spinner 1`] = `
<Styled(div)
  className="test"
  width="50px"
>
  <svg
    className="css-fgjwc2"
    viewBox="0 0 66 66"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="33"
      cy="33"
      r="31"
      stroke="url(#gradient)"
      style={
        Object {
          "fill": "transparent",
          "height": "100%",
          "strokeDasharray": 170,
          "strokeDashoffset": 20,
          "strokeWidth": 2,
          "width": "100%",
        }
      }
    />
    <linearGradient
      id="gradient"
    >
      <stop
        offset="50%"
        stopColor="#ffffff"
        stopOpacity="1"
      />
      <stop
        offset="65%"
        stopColor="#ffffff"
        stopOpacity="1"
      />
      <stop
        offset="100%"
        stopColor="#ffffff"
        stopOpacity="0"
      />
    </linearGradient>
    <circle
      cx="36.8"
      cy="2.3"
      fill="#ffffff"
      r="0.5"
      stroke="#ffffff"
      style={
        Object {
          "strokeDasharray": 170,
          "strokeDashoffset": 20,
        }
      }
    />
  </svg>
</Styled(div)>
`;
